### PR TITLE
Feat: get order

### LINKED ISSUE #29: Search Engine
As a customer, I want to search for something below
- product: by its name, category
- store: by store name

and I want some filter options that I can find exactly product that I want to buy. It can be price range, languages, publishing year...

### PR DESCRIPTION
**Pull request by Dat070601**

### CODE REVIEW

File: BookEcommerce.Models/DAL/Repositories/CartRepository.cs [hunk 1]
```diff
@@ -15,6 +15,7 @@ public CartRepository(DbFactory dbFactory) : base(dbFactory)
         {
         }
 
+
```
Comment (justme2002) [inline BookEcommerce.Models/DAL/Repositories/CartRepository.cs]: remove empty line

File: BookEcommerce.Models/DAL/Repositories/PagedList.cs [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+        public bool HasPrevious => CurrentPage > 1;
+        public bool HasNext => CurrentPage < TotalPages;
+
+        public PagedList(List<T> items, int count, int pageNumber, int pageSize)
+        {
+            TotalCount = count;
+            PageSize = pageSize;
+            CurrentPage = pageNumber;
+            TotalPages = (int)Math.Ceiling(count / (double)pageSize);
+
```
Comment (justme2002) [inline BookEcommerce.Models/DAL/Repositories/PagedList.cs:25]: empty line ?

File: BookEcommerce.Models/DAL/Repository.cs [hunk 3]
```diff
@@ -56,5 +57,17 @@ public void Update(T entity)
         {
             DbSet.Update(entity);
         }
+
+        public async Task<List<T>> Find(string keywordSearch, string keywordExist)
+        {
+            var l = dbFactory.DbContext;
```
Comment (justme2002) [inline BookEcommerce.Models/DAL/Repository.cs]: what does this 'l' mean ?

File: BookEcommerce.Models/DAL/Repositories/CartRepository.cs (hunk 1 above)
Comment (Dat070601) [reply BookEcommerce.Models/DAL/Repositories/CartRepository.cs]: I will remove it

File: BookEcommerce.Models/DAL/Repositories/PagedList.cs (hunk 2 above)
Comment (Dat070601) [reply BookEcommerce.Models/DAL/Repositories/PagedList.cs:25]: I will remove it

File: BookEcommerce.Models/DAL/Repository.cs (hunk 3 above)
Comment (Dat070601) [reply BookEcommerce.Models/DAL/Repository.cs]: I will fix it

File: BookEcommerce.Services/Interfaces/IProductService.cs [hunk 4]
```diff
@@ -11,6 +13,7 @@ namespace BookEcommerce.Services.Interfaces
 {
     public interface IProductService
     {
+        PagedList<Product> GetProducts(ProductParameters ptoductParameters);
         Task<ProductResponse> AddProduct(ProductRequest req, string Token);
```
Comment (justme2002) [inline BookEcommerce.Services/Interfaces/IProductService.cs]: Check convention in  parameter 'Token'

Comment (Dat070601) [reply BookEcommerce.Services/Interfaces/IProductService.cs]: yes, I wil

**Review by justme2002 — APPROVED** [review]

**Pull request merged**